realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -226,9 +228,8 @@ protected void removeHandler() {
      * the last transaction was committed.
      *
      * @param destination file to save the Realm to.
-     * @throws java.io.IOException if any write operation fails.
      */
-    public void writeCopyTo(File destination) throws java.io.IOException {
+    public void writeCopyTo(File destination) {
```

Comment:
I would probably remove RealmIOException. I don't see it having any value over a normal IOException. I think this is a case where Java treats IO exceptions as checked, while C++ does not. IMO we should stick with Java conventions.